Imagine/Filter/Loader/UpscaleFilterLoader.php

```diff
@@ -35,7 +35,7 @@ public function load(ImageInterface $image, array $options = array())
 
             $ratio = $widthRatio > $heightRatio ? $widthRatio : $heightRatio;
 
-            $filter = new Resize(new Box($origWidth * $ratio, $origHeight * $ratio));
+            $filter = new Resize(new Box(round($origWidth * $ratio), round($origHeight * $ratio)));
```

Comment:
Yes. The problem is that if we just pass the float value to the Box object it will be casted in the Box constructor which will lead to this issue that float(201) become float(200) which is not what i configured in the filter.